aten/src/ATen/core/jit_type.h

```diff
@@ -324,25 +324,50 @@ inline c10::optional<T> merge_primitive(
   return c10::optional<T>{};
 }
 
-// `VaryingShape` tracks if individual dimensions or a rank vary across
-// profiled runs. A *varying* or *dynamic* dimension is expressed as
-// an empty c10::optional in `sizes_`. If a rank is dynamic, the entire
-// `sizes_` becomes the empty optional.
+struct CAFFE2_API ShapeSymbol {
+    ShapeSymbol(int64_t val, bool statik = true): value_(val), statik_(statik) {}
+
+    explicit operator int() const { return value_; }
+    ShapeSymbol(const ShapeSymbol& ss) {
+      statik_ = ss.statik_;
+      value_ = ss.value_;
+    }
+    ShapeSymbol(): value_(-1), statik_(false) {}
+    static ShapeSymbol getInvalidSymbol() {return ShapeSymbol(); }
+    int64_t value_;
```

Comment:
I don't think these should be public. This is suppose to be an abstract class representing a symbol, not a containing for our particular encoding.